doc/api-extensions.md

```diff
@@ -2158,3 +2158,9 @@ This adds the following new configuration options for ZFS storage pools:
 * `volume.zfs.block_mode`
 * `volume.block.mount_options`
 * `volume.block.filesystem`
+
+## `instance_generation_id`
+
+Adds support for instance generation ID. The VM or container generation ID will change whenever the instance's place in time moves backwards. This allows for the VM guest OS or container to reinitialize any state it needs to avoid duplicating potential state that has already occurred:
```

Comment:
We should say here and in the key documentation below, that at this time, the generation ID is only exposed through to VM type instances.

Reply:
@tomponline actually that's not true as of now : 
```go
"volatile.uuid.generation": validate.Optional(validate.IsUUID),
```
is part of `InstanceConfigKeysAny`.. Should I put it only in `InstanceConfigKeysVM` then ?

Reply:
No, leave it as is. You're misunderstanding me.
We are generating the `voliatile.uuid.generation` ID for VMs and containers.
But we are only passing it through into VMs as the VM generation ID.

The documentation should note this important detail.

Reply:
ah yes I understand